Godeps/_workspace/src/k8s.io/kubernetes/pkg/apis/extensions/validation/validation.go

```diff
@@ -419,6 +419,10 @@ func ValidateGeneratedSelector(obj *extensions.Job) field.ErrorList {
 	allErrs = append(allErrs, apivalidation.ValidateHasLabel(obj.Spec.Template.ObjectMeta, field.NewPath("spec").Child("template").Child("metadata"), "controller-uid", string(obj.UID))...)
 	allErrs = append(allErrs, apivalidation.ValidateHasLabel(obj.Spec.Template.ObjectMeta, field.NewPath("spec").Child("template").Child("metadata"), "job-name", string(obj.Name))...)
 	expectedLabels := make(map[string]string)
+	// Expect all the labels that are actually on the template
```

Comment:
Since there's no clear agreement on this upstream issue, I'd rather drop this. My guts tell me we might end-up with a nasty carry.